src/context_store/lifecycle/purger.py

```
@@ -37,9 +37,19 @@ def __init__(
         graph: GraphAdapter | None,
         retention_days: int,
     ) -> None:
+        # retention_days のバリデーション: 0 以上の整数であることを保証
+        try:
+            val = int(retention_days)
+            if val < 0:
+                raise ValueError
+        except (ValueError, TypeError):
+            raise ValueError(
+                f"retention_days must be a non-negative integer, got {retention_days!r}"
+            )
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)